src/harness/security_testcase.py

```diff
@@ -171,6 +172,29 @@ def doTestCipher(self, cipher, client_cert=None, client_key=None):
     with CiphersOverload(self._sas, [cipher], client_cert, client_key):
       self.assertRegistered([device_a])
 
+  def doSasTestCipher(self, cipher, client_cert, client_key, client_url):
+    """Does a cipher test as described in SSS tests 1 to 5 specification.
+
+    Args:
+      cipher: the cipher openSSL string name to test.
+      client_cert: SAS client certificate file in PEM format to use.
+      client_key: associated key file in PEM format to use.
+      client_url: base URL of the peer client SAS.
+    """
+
+    # Using pyOpenSSL low level API, does the SAS UUT server TLS session checks.
+    self.assertTlsHandshakeSucceed(self._sas_admin._base_url, [cipher],
+                                   client_cert, client_key)
+
+    # Does a regular SAS registration
+    self.SasReset()
+    certificate_hash = getCertificateFingerprint(client_cert)
+    self._sas_admin.InjectPeerSas({'certificateHash': certificate_hash,
+                                   'url': client_url})
+    self._sas_admin.TriggerFullActivityDump()
+    with CiphersOverload(self._sas, [cipher], client_cert, client_key):
+      self._sas.GetFullActivityDump(client_cert, client_key)
```

Comment:
This method GetFullActivityDump would also not use the client_url. I think we should either modify these methods take the URL as input or have the functions use the sas.cfg file and not have the SAS_TH_URL defined and passed as an argument from the test case